course1/15.goroutines/6.goroutine_sync/task1.15.6.5/main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"fmt"
+
+	"strconv"
+	"strings"
+
+	"sync"
+
+	"time"
+)
+
+// User struct implements "user" object with two fields:
+// ID (type int),
+// Name (type string)
+type User struct {
+	ID   int
+	Name string
+}
+
+// Cache struct implements "cache" object with two fields:
+// mu (type sync.RWMutex) for concurrent operations security
+// items (type map[string]interface{}) as a storage for objects.
+type Cache struct {
+	mu    sync.RWMutex
+	items map[string]interface{}
+}
+
+// NewCache generates and return a new Cache object with initialized `items` map
+func NewCache() *Cache {
+	return &Cache{
+		items: make(map[string]interface{}),
+	}
+}
+
+// Set adds an object to the cache using the specified key. Uses Lock to provide concurrent write security.
+func (c *Cache) Set(key string, item interface{}) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.items[key] = item
+}
+
+// Get returns an object from cache of the specified key. Uses RLock to provide concurrent read security.
+func (c *Cache) Get(key string) interface{} {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	return c.items[key]
+}
+
+// keyBuilder concatenates strings given as args, separating them with `:`
+func keyBuilder(keys ...string) string {
+	return strings.Join(keys, ":")
+}
+
+// GetUser converts interface{} type to *User type. The object retrieved from the cache is expected to be of type *User.
+func GetUser(i interface{}) *User {
+	return i.(*User)
+}
+
+// main creates a new Cache object, runs 100 goroutines adding 100 `User` objs to the Cache, runs 100 goroutines
```

Comment:
why so many comments in a file?

Reply:
there was a requrement in the task to give the code a detailed explanation...